### PR TITLE
move some things out of jenkins/

### PR DESCRIPTION
**Pull request by BenTheElder**
- move aws-janitor out of jenkins/
- move e2e image out of jenkins/
  - also move it to a path reflecting the actual published image name
- move kubekins-test image out of jenkins/
  - also move it to a path reflecting the actual published image name

/area images

### CODE REVIEW

File: maintenance/aws-janitor/BUILD [hunk 1]
```diff
@@ -37,3 +37,10 @@ filegroup(
     tags = ["automanaged"],
     visibility = ["//visibility:public"],
 )
+
+go_binary(
+    name = "aws-janitor",
```
Comment (fejta) [inline maintenance/aws-janitor/BUILD]: Did you manually create this? Why is there an aws_janitor go_binary() target?

**Comment by BenTheElder** [comment]
/assign @fejta 
/assign @ixdy 

I'd love to move `bootstrap.py` out too but one will be trickier to do without breaking things and we're now working to eliminate most of what it does anyhow...  oncall.html is probably the only other non-jenkins thing in `jenkins/`.

File: maintenance/aws-janitor/BUILD (hunk 1 above)
Comment (BenTheElder) [reply maintenance/aws-janitor/BUILD]: No, I ran hack/update-bazel.sh, not sure why it kept the old one. Fixing...

Comment (ixdy) [reply maintenance/aws-janitor/BUILD]: probably cause the binary name changed, and it didn't recognize the old one as being autogenerated.

Comment (BenTheElder) [reply maintenance/aws-janitor/BUILD]: done.

**Comment by ixdy** [comment]
I think you're maybe missing a rename fix in the aws-janitor Makefile.

File: maintenance/aws-janitor/BUILD [hunk 2]
```diff
@@ -37,3 +37,4 @@ filegroup(
     tags = ["automanaged"],
     visibility = ["//visibility:public"],
 )
+
```
Comment (ixdy) [inline maintenance/aws-janitor/BUILD]: you should make your editor run buildifier on write. :)

Comment (BenTheElder) [reply maintenance/aws-janitor/BUILD]: nobody has a good extension for bazel yet, i'm sure there's a way to configure run on save though

**Comment by BenTheElder** [comment]
> I think you're maybe missing a rename fix in the aws-janitor Makefile.

fixed

File: maintenance/aws-janitor/Makefile [hunk 3]
```diff
@@ -16,7 +16,7 @@ IMAGE ?= gcr.io/k8s-test-infra-aws/aws-janitor
 VERSION ?= 0.3
 
 image:
-	CGO_ENABLED=0 go build k8s.io/test-infra/jenkins/aws_janitor
+	CGO_ENABLED=0 go build k8s.io/test-infra/jenkins/aws-janitor
```
Comment (ixdy) [inline maintenance/aws-janitor/Makefile]: `s/jenkins/maintenance`?

Comment (BenTheElder) [reply maintenance/aws-janitor/Makefile]: rah, fixed 😕

**Comment by ixdy** [comment]
lgtm2, but we should probably wait till morning to unhold.

**Comment by ixdy** [comment]
(I'm half-expecting tide to merge #6249 before then)

**Comment by krzyzacy** [comment]
lgtm3

**Pull request merged**